### DISCUSSION TITLE
failed to EGL with glad

### DISCUSSION THREAD

**Discussion by hiyyg** [bulletphysics/bullet3#3737] [General]
Hello, I tried to run the egl renderer example `examples/pybullet/examples/testrender_egl.py`,
but I got this error:
`failed to EGL with glad.`
Do you know how to solve this?

**Comment by BlGene** [comment]
Probably:
1. No EGL library (libEGL.so.1, libEGL.so), try `ldconfig -p | grep EGL`
2. No default display eglGetCurrentDisplay()/eglGetDisplay(EGL_DEFAULT_DISPLAY)
(https://github.com/Dav1dde/glad/issues/297)

**Comment by hiyyg** [comment]
I can use the EGL example provided by https://github.com/mmatl/pyrender. I think there is some other problem.

**Comment by COST-97** [comment]
Hello:
   The same error `failed to EGL with glad`, though `show_gui=False.`
The output of the commands
`cd calvin_env/egl_check
bash build.sh  # should have been built automatically, but try running this again
python list_egl_options.py`
is:
`
----------Default-------------
Starting EGL query
Loaded EGL 1.5 after reload.
GL_VENDOR=Mesa/X.org
GL_RENDERER=llvmpipe (LLVM 12.0.0, 256 bits)
GL_VERSION=4.5 (Core Profile) Mesa 21.2.6
GL_SHADING_LANGUAGE_VERSION=4.50
Completeing EGL query
b'EGL device choice: -1 of 1.\n'
number of EGL devices: 1
----------Option #1 (id=0)-------------
Starting EGL query
EGL device choice: 0 of 1 (from EGL_VISIBLE_DEVICE)
Loaded EGL 1.5 after reload.
GL_VENDOR=Mesa/X.org
GL_RENDERER=llvmpipe (LLVM 12.0.0, 256 bits)
GL_VERSION=4.5 (Core Profile) Mesa 21.2.6
GL_SHADING_LANGUAGE_VERSION=4.50
Completeing EGL query
`

The output of the commands
`ldconfig -p | grep libEGL`
is
`        libEGL_mesa.so.0 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libEGL_mesa.so.0
        libEGL.so.1 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libEGL.so.1
        libEGL.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libEGL.so`

nvidia driver:
`NVRM version: NVIDIA UNIX x86_64 Kernel Module  525.105.17  Tue Mar 28 18:02:59 UTC 2023
GCC version:  gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC)`

Could you give me some advice? Thanks!

**Reply by MyhooSalsa** [reply]
Hi, have you fixed it out?
